buffer/src/main/java/io/netty/buffer/CompositeByteBuf.java

```diff
@@ -101,6 +102,9 @@ public CompositeByteBuf(
         }
 
         int len = buffers instanceof Collection ? ((Collection<ByteBuf>) buffers).size() : 0;
+        if (buffers instanceof Unpooled.WrappingIterable) {
+            len = ((Unpooled.WrappingIterable<?>) buffers).maxSize();
+        }
```

Comment:
Should we merge the two instanceof things ?

```
if (buffers instanceof Unpooled....) {
   len = ...
} else if (buffers instanceof Collection) {
    len = ...;
} else {
    len = 0;
}
```

Reply:
@njhill did you miss this one ?

Reply:
@normanmaurer no I didn't miss it, check the changes tab/commits ... I think it's a github quirk making it show the old version here still